modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelFactory.java

```diff
@@ -0,0 +1,453 @@
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("Loading artifact descriptor from '{}'..." + jsonFile.getAbsolutePath());
+    }
+
+    try (InputStream stream = new BufferedInputStream(new FileInputStream(jsonFile))) {
+      return IOUtils.toString(stream);
+    } catch (IOException e) {
+      throw new IllegalArgumentException(format("Could not read extension describer on artifact '%s'",
+                                                jsonFile.getAbsolutePath()),
+                                         e);
+    }
+  }
+
+  private BundleDescriptor buildBundleDescriptor(ArtifactCoordinates artifactCoordinates) {
+    return new BundleDescriptor.Builder()
+        .setArtifactId(artifactCoordinates.getArtifactId())
+        .setGroupId(artifactCoordinates.getGroupId())
+        .setVersion(artifactCoordinates.getVersion())
+        .setBaseVersion(artifactCoordinates.getVersion())
+        .setType(artifactCoordinates.getType())
+        .setClassifier(artifactCoordinates.getClassifier())
+        .build();
+  }
+
+  private void getExportedPackagesAndResources() throws IOException {
+    // TODO: get these fields from mule-artifact.json if classLoaderModelDescriptor field is present there, and in that case check
+    // that the packages and resources effectively exist
+    // TODO: search the POM for a changed source directory under the tag <sourceDirectory>
+    Path javaDirectory = get(projectFolder.getAbsolutePath(), "src/main/java");
+    // look for all the sources under the java directory
+    List<Path> allJavaFiles = Files.walk(javaDirectory)
+        .filter(Files::isRegularFile)
+        .collect(toList());
+    Predicate<Path> isJavaFile = path -> FilenameUtils.getExtension(path.toString()).endsWith(JAVA_EXTENSION);
+    exportedPackages = allJavaFiles.stream()
+        .filter(isJavaFile)
+        .map(path -> {
+          Path parent = javaDirectory.relativize(path).getParent();
+          // if parent is null, it implies "default package" in java, which means we need an empty string for the
+          // exportedPackages
+          return parent != null ? parent.toString() : DEFAULT_PACKAGE_EXPORT;
+        })
+        .map(this::escapeSlashes)
+        .map(s -> s.replace(CLASS_PATH_SEPARATOR, PACKAGE_SEPARATOR))
+        .distinct()
+        .collect(toList());
+
+    // TODO: search the POM for resource directories under <resources> tag and search in src/main/mule too
+    Path resourcesDirectory = get(projectFolder.getAbsolutePath(), "src/main/resources");
+    // look for all the sources under the resources directory
+    List<Path> allResourcesFiles = Files.walk(resourcesDirectory)
+        .filter(Files::isRegularFile)
+        .collect(toList());
+    exportedResources = allResourcesFiles.stream()
+        .map(resourcesDirectory::relativize)
+        .map(Path::toString)
+        .map(this::escapeSlashes)
+        .collect(toList());
+
+    // TODO: add test resources
```

Comment:
i still need to understand why the packager did this in the first place...

Reply:
Created user story [W-11203142](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000xny0KYAQ/view) to tackle it.